test-galasactl-local.sh

```diff
@@ -0,0 +1,267 @@
+                                          buildTool="$1"
+                                          ;;
+        -h | --help )                     usage
+                                          exit
+                                          ;;
+        * )                               error "Unexpected argument $1"
+                                          usage
+                                          exit 1
+    esac
+    shift
+done
+
+if [[ "${binary}" != "" ]]; then
+    case ${binary} in
+        galasactl-darwin-amd64 )            echo "Using the galasactl-darwin-amd64 binary"
+                                            ;;
+        galasactl-darwin-arm64 )            echo "Using the galasactl-darwin-arm64 binary"
+                                            ;;
+        galasactl-linux-amd64 )             echo "Using the galasactl-linux-amd64 binary"
+                                            ;;
+        galasactl-linux-s390x )             echo "Using the galasactl-linux-s390x binary"
+                                            ;;
+        galasactl-windows-amd64.exe )       echo "Using the galasactl-windows-amd64.exe binary"
+                                            ;;
+        * )                                 error "Unrecognised galasactl binary ${binary}"
+                                            usage
+                                            exit 1
+    esac
+else
+    error "Need to specify which binary of galasactl to use."
+    usage
+    exit 1  
+fi
+
+if [[ "${buildTool}" != "" ]]; then
+    case ${buildTool} in
+        maven  )            echo "Using Maven"
+                            ;;
+        gradle )            echo "Using Gradle"
+                            ;;
+        * )                 error "Unrecognised build tool ${buildTool}"
+                            usage
+                            exit 1
+    esac
+else
+    error "Need to specify which build tool to use to build the generated project."
+    usage
+    exit 1  
+fi
+
+#--------------------------------------------------------------------------
+# Initialise Galasa home
+function galasa_home_init {
+    h2 "Initialising galasa home directory"
+
+    rm -rf ${BASEDIR}/temp
+    mkdir -p ${BASEDIR}/temp
+    cd ${BASEDIR}/temp
+
+    cmd="${BASEDIR}/bin/${binary} local init --development \
+    --log -"
+
+    info "Command is: $cmd"
+
+    $cmd
+    rc=$?
+    if [[ "${rc}" != "0" ]]; then 
+        error "Failed to initialise galasa home"
+        exit 1
+    fi
+    success "Galasa home initialised"
+}
+
+#--------------------------------------------------------------------------
+# Invoke the galasactl command to create a project.
+function generate_sample_code {
+    h2 "Invoke the tool to create a sample project."
+
+    cd ${BASEDIR}/temp
+
+    export PACKAGE_NAME="dev.galasa.example.banking"
+    ${BASEDIR}/bin/${binary} project create --package ${PACKAGE_NAME} --features payee --obr --${buildTool} --force --development --log -
```

Comment:
Should we be using the --maven or --gradle flags, based on the mode the script is running in ?